htdocs/product/stock/tpl/stockcorrection.tpl.php

```diff
@@ -128,26 +144,34 @@ function init_price()
 }
 
 // Serial / Eat-by date
-if (!empty($conf->productbatch->enabled) &&
+if (ismodEnabled('productbatch') &&
 (($object->element == 'product' && $object->hasbatch())
 || ($object->element == 'stock'))
 ) {
 	print '<tr>';
 	print '<td'.($object->element == 'stock' ? '' : ' class="fieldrequired"').'>'.$langs->trans("batch_number").'</td><td colspan="3">';
-	print '<input type="text" name="batch_number" class="minwidth300" value="'.GETPOST("batch_number").'">';
+	if ($pdluoid > 0) {
+		// If form was opened for a specific pdluoid, field is disabled
+		print '<input type="text" name="batch_number_bis" size="40" disabled="disabled" value="'.(GETPOST('batch_number') ?GETPOST('batch_number') : $pdluo->batch).'">';
+		print '<input type="hidden" name="batch_number" value="'.(GETPOST('batch_number') ?GETPOST('batch_number') : $pdluo->batch).'">';
+	} else {
+		print '<input type="text" name="batch_number" class="minwidth300" value="'.(GETPOST('batch_number') ? GETPOST('batch_number') : $pdluo->batch).'">';
+	}
 	print '</td>';
 	print '</tr>';
 	print '<tr>';
 	if (empty($conf->global->PRODUCT_DISABLE_SELLBY)) {
 		print '<td>'.$langs->trans("SellByDate").'</td><td>';
 		$sellbyselected = dol_mktime(0, 0, 0, GETPOST('sellbymonth'), GETPOST('sellbyday'), GETPOST('sellbyyear'));
-		print $form->selectDate($sellbyselected, 'sellby', '', '', 1, "");
+		// If form was opened for a specific pdluoid, field is disabled
+		print $form->selectDate((!empty($sellbyselected) ? $sellbyselected : $pdluo->sellby), 'sellby', '', '', 1, "", 1, 0, ($pdluoid > 0 ? 1 : 0));
```

Comment:
May be 
`print $form->selectDate($pdluo->id > 0 ? $pdluo->sellby : $sellbyselected), 'sellby', '', '', 1, "", 1, 0, ($pdluoid > 0 ? 1 : 0));
`instead of 
`print $form->selectDate((!empty($sellbyselected) ? $sellbyselected : $pdluo->sellby), 'sellby', '', '', 1, "", 1, 0, ($pdluoid > 0 ? 1 : 0));`

Reply:
@eldy done